File: 0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> bigList = new ArrayList<>();
        Arrays.sort(nums);
        int prevI = Integer.MIN_VALUE;
        
        for(int i=0;i<nums.length;i++){
            if(nums[i] != prevI){
                int j = i+1;
                int k = nums.length-1;
                int prevJ = Integer.MIN_VALUE;
                int prevK = Integer.MAX_VALUE;
                while(j<k){
                    if(nums[j] == prevJ){
                        j++;
                    }
                    else if(nums[k] == prevK){
                        k--;
                    }
                    else{
                        if(nums[i] + nums[j] + nums[k] == 0){
                            prevI = nums[i];
                            prevJ = nums[j];
                            prevK = nums[k];
                            List<Integer> list = new ArrayList<>();
                            list.add(nums[i]);
                            list.add(nums[j]);
                            list.add(nums[k]);
                            bigList.add(list);
                            j++;
                            k--;
                        }
                        else if(nums[i] + nums[j] + nums[k] > 0){
                            k--;
                        }
                        else if(nums[i] + nums[j] + nums[k] < 0){
                            j++;
                        }
                    }
                    
                }
            }
            
        }
        return bigList;
    }
}

File: 0015-3sum/0015-3sum.go

func threeSum(nums []int) [][]int {
    var bigList = [][]int{}
    previ := math.MinInt64
    sort.Ints(nums)
    for i:=0;i<len(nums);i++{
        if(nums[i] != previ){
            j:=i+1
            k:=len(nums)-1
            prevj := math.MinInt64
            prevk := math.MaxInt64

            for j<k{
                if(prevj == nums[j]){
                    j++
                }else if(prevk == nums[k]){
                    k--
                }else{
                    if(nums[i] + nums[j] + nums[k] == 0){
                        var list = make([]int,3)
                        
                        list[0] = nums[i]
                        list[1] = nums[j]
                        list[2] = nums[k]
                        
                        previ = nums[i]
                        prevj = nums[j]
                        prevk = nums[k]
                        bigList = append(bigList,list)
                        j++
                        k--
                    }else if(nums[i] + nums[j] + nums[k] > 0){
                        k--
                    }else if(nums[i] + nums[j] + nums[k] < 0){
                        j++
                    }
                }
            }    
        }
    }
    return bigList
}